### DISCUSSION TITLE
How do I change the status code of http redirection to https from 308 to 301?

### DISCUSSION THREAD

**Discussion by franksun001** [vercel/community#1286] [Help] [answered]
### Summary

<img width="772" alt="image" src="https://user-images.githubusercontent.com/29445563/213085304-26f0acee-e2ef-41d4-9452-79b0a120b103.png">
<img width="528" alt="企业微信截图_ffa8d6ea-d1a2-4c64-80fa-fb37a5e5f798" src="https://user-images.githubusercontent.com/29445563/213085702-c81d15b8-9c0a-4d29-a240-89b04f87ddd4.png">

I deploy our Nextjs project to Vercel, can I change the status code from 308 to 301 ?

Thanks!!!

**Comment by amyegan** [comment, marked as answer]
A 308 is generally preferred since the exact same request will be performed for the new location versus a 301 request in which the request could be altered. 

Can you tell us more about why you want to use a [301](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/301) instead of a [308](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/308) for HTTP to HTTPS redirects? Maybe there is already a solution available that will work for your situation.

**Reply by preslavnolimit** [reply]
SEO agencies might push that.

**Comment by ksnyder** [comment]
I have the same problem and it doesn't matter to me whether it's 301 or 308, but Google Analytics will only work to recognize a new domain name if the old one sends a 301 to the new one.  And, when I configure my redirect on Vercel, it gives a choice of 4 different 30x codes -- I choose 301, but in fact you are sending a 308 instead -- why?   I've attached the error from Google:  <img width="672" alt="Screenshot 2023-09-07 at 9 12 24 AM" src="https://github.com/vercel/community/assets/182565/287e3969-3f86-4b7b-a1d4-160c0b6fe957">

**Reply by amyegan** [reply]
Hey there. You can find a more detailed explanation of why `308` is used in https://github.com/vercel/community/discussions/3972#discussioncomment-6944372

**Reply by ksnyder** [reply]
Yes, but there's still the question of how to deal with Google Search's requirement that a domain move happen with a 301.

**Reply by amyegan** [reply]
When [redirecting a custom domain](https://vercel.com/docs/projects/domains/deploying-and-redirecting#redirecting-domains) to a new domain, you can set the status code in project settings. If you use the [redirects](https://vercel.com/docs/projects/project-configuration#redirects) property in `vercel.json`, you can use `"permanent": true` or `"statusCode": 301` to give it a permanent redirect status code. You can also set the status code on [dynamic redirects](https://vercel.com/docs/edge-network/redirects#dynamic-redirects) from serverless functions and edge functions.

The `308` status code is used for http to https redirects for the reason listed above https://github.com/vercel/community/discussions/1286#discussioncomment-4730480 and explained in https://github.com/vercel/community/discussions/3972#discussioncomment-6944372

**Comment by bernardogarza** [comment]
We are having the same issue
